src/shared/string_op.c

```diff
@@ -759,6 +760,54 @@ long w_parse_time(const char * string) {
     return seconds >= 0 ? seconds : -1;
 }
 
+// Get time unit from seconds
+
+char*  w_seconds_to_time_unit(long seconds, bool long_format) {
+
+    if (seconds < 0) {
+        return "invalid";
+    }
+    else if (seconds >= 604800) {
+        return long_format ? "week(s)" : "w" ;
+    }
+    else if (seconds >= 86400) {
+        return long_format ? "day(s)" : "d" ;
+    }
+    else if (seconds >= 3600) {
+        return long_format ? "hour(s)" : "h" ;
+    }
+    else if (seconds >= 60) {
+       return long_format ? "minute(s)" : "m" ;
+    }
+    else  {
+       return long_format ? "second(s)" : "s" ;
+    }
+}
+
+// Get time value from seconds
+
+long w_seconds_to_time_value(long seconds) {
+    
+    if(seconds < 0) {
+        return -1;
+    }
+    else if (seconds >= 604800) {
```

Comment:
I suggest also define these constants in a header file.

Reply:
Done